ccm_web/server/src/canvas/canvas.interfaces.ts

```diff
@@ -145,11 +202,15 @@ export interface CanvasErrorBody {
 }
 
 export function isCanvasErrorBody (value: unknown): value is CanvasErrorBody {
-  if (!hasKeys(value, ['errors']) || !Array.isArray(value.errors)) {
+  if (!hasKeys(value, ['errors'])) {
     return false
-  } else {
+  }
+
+  if (Array.isArray(value.errors)) {
     const result = value.errors.map(e => isCanvasError(e)).every(e => e)
     return result
+  } else {
+    return true
```

Comment:
If it's not an array, what type is it? Is it one object with a key of `message`? If so, this needs to be updated to be
```
} else {
  return isCanvasError(value.errors)
```
and `CanvasErrorBody` needs to be
```
interface CanvasErrorBody {
  errors: CanvasError | CanvasError[]
}
```
and then likely `handleAPIError` needs more logic.

Returning `true` just leads to the type system assuming it has the old shape, which will likely lead to runtime errors.

Reply:
@lsloan, would you like help with this? This could also be handled separately as part of #169.

Reply:
I'm pretty sure it was not saying that "the response was not an error", just that it couldn't parse the error body shape returned because it hasn't been taught to recognize it. The error code should still be passed through.

Honestly, the error returned for this particular case is pretty messy; something like this:
```
{"errors":{"user":{"pseudonyms":[{"attribute":"pseudonyms","type":"invalid","message":"invalid"}]},"pseudonym":{"unique_id":[{"attribute":"unique_id","type":"taken","message":"ID already in use for this account and authentication provider"}]},"observee":{},"pairing_code":{},"recaptcha":null}}
```
It'd probably be ideal to pull the `message` out here, but we'd have to teach TypeScript to recognize data of this shape in addition to the other more standard pattern I have already handled.

If it was just handling `{ "errors": { "message": "some string" } }`, which is what I thought was missing from #169, I'd probably do something like this: https://github.com/ssciolla/canvas-course-manager-next/commit/90283d2f63740ef204ce3736afa46dabb47b34f5

Reply:
I've taken a stab at improving this in the PR related to checking for a "user already exists" error. I was mistaken about the basic shape first described in #169, but I think that has been corrected. You can test it by commenting out the `IsIn` decorator on one of the enrollment routes, and trying to send an invalid role via Swagger.